### ISSUE TITLE
Fix the type of the certifierCredential property

### ISSUE THREAD

**Issue by llemeurfr** [mattgarrish/epub-revision#2112]
certifierCredential "Identifies a credential or badge that establishes the authority of the party identified in the associated [certifiedBy property](https://www.w3.org/TR/epub-a11y-11/#certifiedBy) to certify content accessible."

EPUB Accessibility 1.1 [specifies that the type of certifierCredential is a string](https://www.w3.org/TR/epub-a11y-11/#certifierCredential). There is no mention of a URL there. 

This type is also visible [in the example displayed in the Display Techniques for EPUB Accessibility Metadata](https://www.w3.org/publishing/a11y/UX-Guide-metadata/techniques/epub-metadata/#certifier-credential).

But in the Display Techniques, a note indicates that "This metadata could be a simple text string in which case you would just display it, or it could be a URI to the certifier’s credential web page.". It is followed by different examples, including one with an image which is never defined (and relies on an undefined out of band distribution of logos ... an interoperability killer). 

At the time we're trying to standardize the display of a11y metadata on library websites and in reading systems, and because this property is not the most important and publishers will have a hard time to understand it, having such a flexible type is in our opinion bad practice: publishers should not be left without precise indication and applications should not have to parse a string to decide how to display a metadata property. 

We therefore ask for a stricter specification of this property type: plain string or URL, but not both.

**Comment by mattgarrish** [comment]
Transferring this to the epub-specs repository, as any changes to the requirements and cardinality has to be done there, not in the CG.

**Comment by llemeurfr** [comment]
I agree that we cannot ban URL at a validation level, but we'd love to see the property specified as a plain language label. 

For the cardinality, it can be discussed. But why would a *certifier* have several *credentials*?
Note: the cardinality I'm speaking of here is the one of the credential vs its relative certifier.  

Thanks Matt.

**Comment by mattgarrish** [comment]
> But why would a _certifier_ have several _credentials_

Credentials are just badges of the person who reviewed the publication, not something the reviewer is assigning to the content. It's like financial professionals listing all their certifications after their name.

It'd be like me saying I'm "Matt Garrish, GCA Certified, DAISY OK Certified, EAA Certified, ...". I can list all those in a publication to make me look more impressive as the person who has checked conformance.

I don't know how many credentials any publisher/reviewer will actually bother to get, but I'm hesitant to limit them to only one.

**Comment by clapierre** [comment]
Currently Benetech's GCA program is providing a URL to our GCA Credential Page where we have a "Mark/Stamp" logo displayed inside the certifiersCredential metadata.

I would love to see a solution here where if the certifier has a Logo/Mark/Stamp that could be displayed which would be a recognizable image something like the "CE" certification.  Not sure if this could be part of the publication or not but this is a bigger discussion.

This way if we had the "Text" would be the alt text of the image, and then either a link to the image or embedded within the EPUB.

We had this issue where VitalSource Bookshelf were willing to display the GCA Logo but we didn't have a great way to do that without matching a specific URL inside the certifierCredential's metadata and providing our logo to them out of band, which I agree isn't ideal or scalable.

**Comment by mattgarrish** [comment]
I'm getting the sense we're not going to solve this in the next week, so maybe we should defer this issue?

It sounds like we have some work to figure out if another property is needed and/or whether the spec or the issuer of the credential should be the one setting the use expectations. Perhaps whatever we resolve on can be implemented in the guide as a way of incubating.

**Comment by llemeurfr** [comment]
If the certifier credential is a unique string, the certifier can send a complete info as "GCA Certified, DAISY OK Certified, EAA Certified". Easy to display.

If the cardinality of the property must really be multiple, this will make te life of library platform developer more complex (should it be displayed as a table ?).  

Re. allowing recognizable images (whatever the solution is), I doubt that library managers would accept to insert in their web pages images they do not control (size, potential hacks), plus design a UI which can handle an undefined number of images (even if some sort of carrousel could work).

As much as we can, let's make the life of library platform developers and reading system developers less miserable :-)

**Comment by mattgarrish** [comment]
Bear in mind that restricting the property at this point isn't within our charter. We can loosen restrictions, but we can't add new restrictions that can invalidate content. I don't know how much the field is being used, but we've avoided other changes like this for the same reason of the unknown.

As for merging metadata, I'm not sure that's an ideal solution. We need to consider that this metadata is intended to translate between onix and marc, too, and having single text fields containing multiple values may not be desired for systems that expect formal metadata. It'd also put this field at odds with all the other metadata in the package. You can merge metadata if you only care about display, but I don't think we want to set that as a precedent here that it's what you must do.

I also thought Charles was wanting to use URLs to identify when to use an icon, not that you'd grab one from the URL, but I could be wrong.

Regardless, rushing changes has a way of coming back to bite us, but I'll defer to @avneeshsingh on whether he wants to take this up in the remaining four days next week. We are tight to the CR deadline now.

**Comment by clapierre** [comment]
At this point I agree let's leave this for now and address this in a future 1.2 version.  Where we have potentially separate text string for the Name of the certifier's credential and a 2nd piece of metadata indicating its "Badge/Mark" image.

**Comment by llemeurfr** [comment]
For the record, Readium toolkits will process certifier credential as a string (which is consistent with the current draft of https://www.w3.org/TR/epub-a11y-11/#certifierCredential). In the Test Applications of the toolkits and in Thorium Reader, a URL will be displayed as-is, i.e. not clickable.
These toolkits will consider 0..1 credential per certifier (we interpret the 0..n cardinality of the property as "N possible if there are N certifiers").

**Comment by LaurenceZaysser** [comment]
To me, a `certifierCredential `refers to everything giving legitimacy to the body in charge of the evaluation. This includes training, diplomas, and even process.
I am afraid that plain text would be too harsh to reassure prospect customers. Ideally, its value should be HTML (including paragraphs, lists, URL), instead of plain text.
Here is an example of Rustici credentials for his audits: [https://rusticisoftware.com/security-certifications/?utm_source=google&utm_medium=natural_search](https://rusticisoftware.com/security-certifications/?utm_source=google&utm_medium=natural_search)
I would treat `certifierCredential `as the `certifierReport `in this case.
```
<meta
    property="a11y:certifiedBy"
    id="certifier">
   Accessibility Testers Group
</meta>
<link
    rel="a11y:certifierReport"
    refines="#certifier"
    href="https://example.com/a11y-report/"/>
<link
    rel="a11y:certifierCredential"
    refines="#certifier"
    href="https://example.com/a11y-tester-group-credential/"/>
```

**Comment by mattgarrish** [comment]
We discussed this at the last task force meeting and the group was leaning to recommending text labels. The experience from the display guide was that URLs without a readable name are not helpful, and EPUB's metadata is too ugly to model an ideal solution.

@clapierre I don't think this would change things for you since we can't account for images, either. It's still going to require a separate process to match the value of the credential and display the logo. But we wanted to wait until you were back before trying to find a consensus position.

The advantage of a named credential is for all the systems that might encounter your badge and not know what it is, they can still display a readable label.

While there are advantages to having linkable URLs, making these available to users also comes with security risks. Asking the user to search for public credentials they don't recognize shouldn't be too difficult, assuming they have that much interest. The people who care most about credentials are likely already going to know what these mean.

**Comment by iherman** [comment]
This was discussed during the [pmwg meeting on 17 July 2025](https://w3c.github.io/pm-wg/minutes/2025-07-17-a11y.html#20cb).

<details><summary><i>View the transcript</i></summary>

<h3>Fix the type of the certifierCredential property: Issue #2112</h3>
<p><cite>&lt;AvneeshSingh&gt;</cite> <a href="https://github.com/w3c/epub-specs/issues/2112" rel="noopener noreferrer">w3c/<wbr>epub-specs#2112</a></p>
<p><cite>AvneeshSingh:</cite> about certifier credential should it be a text string, URLs in it?</p>
<p><cite>AvneeshSingh:</cite> , link is important for GCA logo.</p>
<p><cite>&lt;gautierchomel&gt;</cite> CharlesL: when we fisrt come with this metadata it was supposed to be a link to a credencial page with more information including logo. Distributors like Vital Source, we'v negociated and if they see this url, the display the logo. If we change to text only, thousands of EPUBs out there will have an issue and need to be updated.</p>
<p><cite>AvneeshSingh:</cite> when using this field in the reading systems it may not be useful to the users since they can't display the logo</p>
<p><cite>&lt;gautierchomel&gt;</cite> AvneeshSingh: plain link is not user friendly and not often used as a link by the reading systems. Users would have a better information if they read GCA instead of a raw URL.</p>
<p><cite>&lt;gautierchomel&gt;</cite> CharlesL: One way would be to recomand RS to find an image when they get this specific URL. But yes it's extra work.</p>
<p><cite>AvneeshSingh:</cite> whats the recommendation from the group? status quo?<br>
<span>… what about using refines to provide the URL.</span></p>
<p><cite>Dale:</cite> is this a matter for the Standards, or is this a requirement for the one certification body.</p>
<p><cite>AvneeshSingh:</cite> Thorium wants to display this metadata, if you have the URL to the credential</p>
<p><cite>&lt;gautierchomel&gt;</cite> CharlesL: we could add a new metadata like certifier credential name with a refine, so it can be used by RS without needing to update too many epubs.</p>
<p><cite>AvneeshSingh:</cite> Lets see what Matt things of this approach.</p>
<p><cite>George:</cite> right now there is a link that the RS cannot fetch the logo from the link they can show it but may not be an active link.<br>
<span>… now a new requirement they they want to display this for discovery.</span></p>
<p><cite>Hadrien:</cite> refine in general in EPUB is problematic a RS will probably ignore it. we get a flat representation of the metadata.</p>
<p><cite>George:</cite> but the metadata would be there and therefore just display the name of the "organization".</p>
<p><cite>Hadrien:</cite> We roll out things that potentially won't get used.</p>
<p><cite>George:</cite> we have the new metadata why need for refines?</p>
<p><cite>Hadrien:</cite> part of the problem is multiple certifiers potentially.</p>
<p><cite>AvneeshSingh:</cite> if a RS is showing a link is that a problem. but it may not be clickable.</p>
<p><cite>George:</cite> a lot of links are pretty understandable. but having a link that has GCA in the link is pretty readable.</p>
<p><cite>AvneeshSingh:</cite> we don't want to disrupt existing implementations.</p>
<p><cite>George:</cite> that should be good enough for a reading system to present to end user.</p>

<hr /></details>
